chainHead: Allow single connection calls and enforce order of events (#137)
paritytech/json-rpc-interface-spec: src/api/chainHead.md

These functions are the functions most of the JSON-RPC clients will most commonly use. A JSON-RPC server implementation is encouraged to prioritize serving these functions over other functions, and to put pinned blocks in a quickly-accessible cache.

> In case of a JSON-RPC server that accepts multiple different client connections, the `chainHead` functions must be called from the same connection.

Reviewer: ```suggestion
The `chainHead` functions must be called from the same connection in order to work together.
```

Do we need the beginning bit, because whether multiple or one conenction allowed, the same is true anyway?

And then I just added "in order to work together." because you can of course call them from different connections (where possible), but in that case they won't work together. There's prob a better way to work this overall (ie add an example of what is meant), but I didn't want to dedicate lots of words to this minor point :)

Author: Yep I like this better :D

Reviewer 2: I'm the one who asked for this beginning bit in the comment above. The problem is that "connection" really means nothing. You addressed my concern, but then it got changed again and back to saying "connection".

In your head you probably understand that it means "WebSocket connection or similar", but the phrase as it is written now is senseless.
